vlm/src/main/scala/RasterSourceRDD.scala

```diff
@@ -0,0 +1,105 @@
+package geotrellis.contrib.vlm
+
+
+import geotrellis.raster._
+import geotrellis.vector._
+import geotrellis.spark._
+import geotrellis.spark.partition._
+import geotrellis.spark.tiling._
+import geotrellis.proj4._
+
+import org.apache.spark._
+import org.apache.spark.rdd._
+
+
+object RasterSourceRDD {
+  final val PARTITION_BYTES: Long = 128l * 1024 * 1024
+
+  def apply(
+    sources: Seq[RasterSource],
+    layout: LayoutDefinition,
+    partitionBytes: Long = PARTITION_BYTES
+  )(implicit sc: SparkContext): MultibandTileLayerRDD[SpatialKey] = {
+    val mapTransform = layout.mapTransform
+    val extent = mapTransform.extent
+    val combinedExtents = sources.map { _.extent }.reduce { _ combine _ }
+
+    val cellType = {
+      val cellTypes = sources.map { source => Set(source.cellType) }.reduce { _ ++ _ }
+
+      if (cellTypes.size > 1)
+        throw new Exception(s"All RasterSources must have the same CellType, but multiple ones were found: $cellTypes")
+      else
+        cellTypes.head
+    }
+
+    val crs = {
+      val projections = sources.map { source => Set(source.crs) }.reduce { _ ++ _ }
+
+      if (projections.size > 1)
+        throw new Exception(s"All RasterSources must be in the same projection, but multiple ones were found: $projections")
+      else
+        projections.head
+    }
+
+    val layerKeyBounds = {
+      val bounds =
+        sources.map { source =>
+          val projectedExtent = ProjectedExtent(source.extent, source.crs)
+          val boundsKey = projectedExtent.translate(SpatialKey(0, 0))
```

Comment:
This part is confusing. The CRS of all the sources is required to be the same here and it must be the same CRS as the `LayoutDefinition`. So there is no need to reproject. And `SpatialKey(0,0)` looks like a placeholder.

You should union the source extents and use `KeyBounds(mapTransform(combinedExtents))` directly.